tests/testthat/test-data_modify.R

```diff
@@ -576,6 +595,163 @@ test_that("data_modify .if/.at arguments", {
+  )
+  out2b <- data_modify(
+    iris,
+    sepwid = as_expression("2 * Sepal.Width"),
+    seplen = {"5 * Sepal.Length"}
+  )
+  e <- c("sepwid = 2 * Sepal.Width", "seplen = 5 * Sepal.Length")
+  out3b <- data_modify(iris, as_expression(e))
+  e <- "2 * Sepal.Width"
+  out4b <- data_modify(iris, sepwid = as_expression(e), seplen = 5 * Sepal.Length)
+
+  expect_equal(head(out1b), head(out2b), ignore_attr = TRUE, tolerance = 1e-4)
+  expect_equal(head(out1b), head(out3b), ignore_attr = TRUE, tolerance = 1e-4)
+  expect_equal(head(out1b), head(out4b), ignore_attr = TRUE, tolerance = 1e-4)
+
+  # no expression
+  out <- data_modify(iris, sepwid = "2 * Sepal.Widht")
+  expect_identical(
+    head(out$sepwid),
+    c(
+      "2 * Sepal.Widht", "2 * Sepal.Widht", "2 * Sepal.Widht", "2 * Sepal.Widht",
+      "2 * Sepal.Widht", "2 * Sepal.Widht"
+    )
+  )
+
+  # complex example
+  e <- "2 * Sepal.Width"
+  f <- "half_petal = 0.5 * Petal.Length"
+  a <- "string"
+  num <- 1:5
+  out_complex <- data_modify(
+    iris,
+    sepwid = as_expression(e),
+    seplen = 5 * Sepal.Length,
+    {f},
+    new_var = a,
+    new_num = num,
+    new_var2 = "ho",
+    new_num2 = 4:6,
+    Sepal.Length = NULL,
+    Petal.Length = NULL,
+    Sepal.Width = NULL,
+    Petal.Width = NULL
+  )
+  expect_snapshot(print(head(out_complex)))
+})
+
+
+test_that("data_modify works with new expressions, grouped_df, different use cases same results", {
+  data(efc, package = "datawizard")
+  grouped_efc <- data_group(efc, "c172code")
+  new_efc1 <- data_modify(
+    grouped_efc,
+    c12hour_c = center(c12hour),
+    c12hour_z = c12hour_c / sd(c12hour, na.rm = TRUE),
+    c12hour_z2 = standardize(c12hour),
+    id = 1:n()
+  )
+
+  new_efc2 <- data_modify(
+    grouped_efc,
+    as_expression("c12hour_c = center(c12hour)"),
+    c12hour_z = as_expression("c12hour_c / sd(c12hour, na.rm = TRUE)"),
+    c12hour_z2 = standardize(c12hour),
+    id = 1:n()
+  )
+  expect_equal(head(new_efc1), head(new_efc2), ignore_attr = TRUE, tolerance = 1e-4)
+
+  s <- c(
+    "c12hour_c = center(c12hour)",
+    "c12hour_z = c12hour_c / sd(c12hour, na.rm = TRUE)",
+    "c12hour_z2 = standardize(c12hour)"
+  )
+  new_efc3 <- data_modify(
+    grouped_efc,
+    as_expression(s),
+    id = 1:n()
+  )
+  expect_equal(head(new_efc1), head(new_efc3), ignore_attr = TRUE, tolerance = 1e-4)
+
+  new_efc4 <- data_modify(
+    grouped_efc,
+    c12hour_c = center(c12hour),
+    c12hour_z = as_expression("c12hour_c / sd(c12hour, na.rm = TRUE)"),
+    c12hour_z2 = standardize(c12hour),
+    id = 1:n()
+  )
+  expect_equal(head(new_efc1), head(new_efc4), ignore_attr = TRUE, tolerance = 1e-4)
+})
+
+
+test_that("data_modify errors with new expressions", {
+  e <- "sepwid = 2 * Sepal.Widht"
+  expect_error(
+    data_modify(iris, as_expression(e)),
+    regex = "in the first expression"
+  )
+  expect_error(
+    data_modify(iris, as_expression(e)),
+    regex = "Sepal.Widht"
+  )
```

Comment:
Why do we have two `expect_error()` for each case? If there are several messages to test I'd rather have `expect_snapshot(error = TRUE)`.

Reply:
We have just one error, but this error gives different information in the message:
- which expression fails?
- what is the exact typo?

I wanted to check for both.